File: nas_spec/code_samples/C#/standalone@{id}@collect-data/put.cs
// For more information please refer to https://github.com/checkout/checkout-sdk-net
using Checkout.Sessions;
using Checkout.Sessions.Channel;

ICheckoutApi api = CheckoutSdk.Builder().OAuth()
    .ClientCredentials("client_id", "client_secret")
    .Scopes(OAuthScope.SessionsApp, OAuthScope.SessionsBrowser)
    .Environment(Environment.Sandbox)
    .HttpClientFactory(new DefaultHttpClientFactory())
    .Build();

BrowserSession request = new BrowserSession
{
    AcceptHeader = "Accept:  *.*, q=0.1",
    JavaEnabled = true,
    Language = "FR-fr",
    ColorDepth = "16",
    ScreenWidth = "1920",
    ScreenHeight = "1080",
    Timezone = "60",
    UserAgent =
        "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36",
    ThreeDsMethodCompletion = ThreeDsMethodCompletion.Y,
    IpAddress = "1.12.123.255"
};

try
{
    GetSessionResponse response = await api.SessionsClient().UpdateSession("session_id", request);
}
catch (CheckoutApiException e)
{
    // API error
    string requestId = e.RequestId;
    var statusCode = e.HttpStatusCode;
    IDictionary<string, object> errorDetails = e.ErrorDetails;
}
catch (CheckoutArgumentException e)
{
    // Bad arguments
}
catch (CheckoutAuthorizationException e)
{
    // Invalid authorization
}

File: nas_spec/code_samples/PHP/standalone@{id}@collect-data/put.php
<?php
//For more information please refer to https://github.com/checkout/checkout-sdk-php

use Checkout\\CheckoutApiException;
use Checkout\\CheckoutAuthorizationException;
use Checkout\\CheckoutSdk;
use Checkout\\Environment;
use Checkout\\OAuthScope;
use Checkout\\Sessions\\Channel\\BrowserSession;
use Checkout\\Sessions\\Channel\\ThreeDsMethodCompletion;

$api = CheckoutSdk::builder()->oAuth()
    ->clientCredentials("client_id", "client_secret")
    ->scopes([OAuthScope::$SessionsApp, OAuthScope::$SessionsBrowser])
    ->environment(Environment::sandbox())
    ->build();

$browserSession = new BrowserSession();
$browserSession->accept_header = "Accept:  *.*, q=0.1";
$browserSession->java_enabled = true;
$browserSession->language = "FR-fr";
$browserSession->color_depth = "16";
$browserSession->screen_width = "1920";
$browserSession->screen_height = "1080";
$browserSession->timezone = "60";
$browserSession->user_agent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36";
$browserSession->three_ds_method_completion = ThreeDsMethodCompletion::$y;
$browserSession->ip_address = "1.12.123.255";

try {
    $response = $api->getSessionsClient()->updateSession("session_id", $browserSession);
} catch (CheckoutApiException $e) {
    // API error
    $error_details = $e->error_details;
    $http_status_code = isset($e->http_metadata) ? $e->http_metadata->getStatusCode() : null;
} catch (CheckoutAuthorizationException $e) {
    // Bad Invalid authorization
}